plugins/base/src/main/kotlin/transformers/documentables/DefaultDocumentableMerger.kt

```diff
@@ -1,36 +1,78 @@
 package org.jetbrains.dokka.base.transformers.documentables
 
+import org.jetbrains.dokka.DokkaConfiguration
 import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.model.properties.WithExtraProperties
 import org.jetbrains.dokka.model.properties.mergeExtras
 import org.jetbrains.dokka.plugability.DokkaContext
 import org.jetbrains.dokka.transformers.documentation.DocumentableMerger
 import org.jetbrains.kotlin.utils.addToStdlib.firstNotNullResult
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
 internal object DefaultDocumentableMerger : DocumentableMerger {
+    var dependencyInfo: Map<DokkaConfiguration.DokkaSourceSet, List<DokkaConfiguration.DokkaSourceSet>>? = null //Sorry for that hack
 
     override fun invoke(modules: Collection<DModule>, context: DokkaContext): DModule {
+        this.dependencyInfo = context.getDependencyInfo()
```

Comment:
This is hack that needs to be refactored, functions that are used to merge, are not in scope of object so you cannot easily reference to dokka context from these functions.